File: Katas/src/main/java/solution/FizzBuzz.java
package solution;

public class FizzBuzz {

    private static final String FIZZ_STR = "Fizz";
    private static final String BUZZ_STR = "Buzz";

    public String of(int num) {
        if (num <= 0)
            return "0";

        return _of(num);
    }

    private String _of(int num) {
        String retStr = "";
        if (num % 3 == 0)
            retStr += FIZZ_STR;
        if (num % 5 == 0)
            retStr += BUZZ_STR;
        return retStr.isEmpty() ? String.valueOf(num) : retStr;
    }
}

File: golang/kata/fizzbuzz/fizzbuzz.go
package fizzbuzz

import "strconv"

type FizzBuzz struct {
}

func (f *FizzBuzz) Of(n int) string {
	// Corner case
	if n == 0 {
		return "0"
	}
	return f.of(n)
}

func (f *FizzBuzz) of(n int) string {
	s := ""
	if n%3 == 0 {
		s += "Fizz"
	}
	if n%5 == 0 {
		s += "Buzz"
	}
	if !isEmpty(s) {
		return s
	}
	return strconv.Itoa(n)
}

func isEmpty(s string) bool {
	return len(s) < 0
}
